lib/linter.js

```diff
@@ -192,6 +170,34 @@ var runChecks = function (ast, fullPath, lines, config) {
                 });
             }
         });
+    };
+
+    // Freeze the AST so linters won't accidentally change it
+    Object.freeze(ast);
+
+    // Fetch inline options and merge it with whatever passed
+    inlineOptions = getInlineOptions(ast.root.first);
+    config = merge(config, inlineOptions);
+
+    // Load any additional linters
+    linters = mergeLintersFromConfig(defaultLinters, config);
+
+    // Fetch all single line options before we start the actual linting
+    ast.root.walkComments(function (node) {
+        var option = parseRules(node);
+        var key = Object.keys(option)[0];
+
+        option[key] = merge(option[key], {
+            line: node.source.start.line
+        });
+
+        lineOptions.push(option);
+    });
+
+    checkNode(ast.root);
```

Comment:
Could you add a comment here, explaining why it's necessary (just for future reference).